### PR TITLE
cmake: fix the build of tests

### PR DESCRIPTION
**Pull request by tchaikov**
perfglue/heap_profiler.cc is referncing boost-system somehow, and
this file is compiled when we are using tcmalloc, so added it to
ALLOC_LIBS along with tcmalloc.

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@cbodley again, could you help me on this? thanks =)

i think this change is introduced by the graylog changes in #7190, i failed to reproduce the compilation error spotted by @irq0 . but it popped up in our gitbuilder at last.

**Comment by cbodley** [comment]
@tchaikov do you have a link to the gitbuilder log?

**Comment by tchaikov** [comment]
@cbodley http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-tarball-trusty-amd64-cmake/log.cgi?log=995a6029bbaf6bdf1e5614bb07a2182187e11884 .

**Comment by cbodley** [comment]
thanks @tchaikov. i also can't figure out why boost_system is getting pulled in - a grep for 'system_category' on our source tree came up empty.

i get the feeling the cmake build is getting this heap_profiler stuff horribly wrong. i read through the automake files, and all they do is build a libperfglue.la and add it to LIBOSD/LIBMON/LIBMDS. we should eventually do an audit on this stuff in cmake to make sure we get it right, but i'm fine with merging your fix in the meantime

**Comment by tchaikov** [comment]
@cbodley root caused and fixed it.

**Comment by tchaikov** [comment]
hey @alimaredia could you help review it?

**Comment by alimaredia** [comment]
@mattbenjamin didn't you say we needed to do the two points Kefu mentioned in his previous post?

**Comment by tchaikov** [comment]
pushed for testing at http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-tarball-trusty-amd64-cmake/#origin/wip-fix-cmake, and it's green.

**Comment by cbodley** [comment]
@tchaikov good catch!

@alimaredia those two changes are no longer needed after this fix

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -76,8 +76,7 @@ else()
 endif()
 
 
-set(EXTRALIBS uuid rt dl
-  ${Boost_LIBRARIES} ${Boost_SYSTEM_LIBRARY} ${ATOMIC_OPS_LIBRARIES})
+set(EXTRALIBS uuid rt dl ${Boost_LIBS} ${ATOMIC_OPS_LIBRARIES})
```
Comment (alimaredia) [inline src/CMakeLists.txt]: @tchaikov ${Boost_LIBS} neither in /usr/share/cmake/modules/FindBoost.cmake nor anywhere else in the source tree. It should be changed to ${Boost_LIBRARIES}

**Comment by tchaikov** [comment]
@alimaredia could you take a look again? I just removed `${Boost_LIBS}` .

**Comment by tchaikov** [comment]
tested at http://gitbuilder.sepia.ceph.com/gitbuilder-ceph-tarball-trusty-amd64-cmake/#origin/wip-fix-cmake , green also.

**Pull request merged**